### PR TITLE
Fix some broken test cases

### PR DESCRIPTION
**Pull request by arjo129**
### Fixed bug
The test cases had the following issues:
* `test_ParticipantRegistry.cpp` was not building against clang due to some issues with namespacing of operators.
* `test_RequestLift.cpp` has a potential buffer overflow/invalid memory access issue. (Thanks @mxgrey for suggesting the fixes).

### CODE REVIEW

**Comment by Yadunund** [comment]
Fyi we had `test_IngestItem` fail yesterday. Not sure if there's a simple for it as well. 
https://github.com/open-rmf/rmf_ros2/pull/88#issuecomment-887340011

**Comment by arjo129** [comment]
@Yadunund I've taken a cursory glance at the said test case. I doubt its something that is very trivial as I've been unable to reproduce its failure. Currently, I'm basically working through issues I see when I run the address sanitizer (which there are a lot of).

File: rmf_traffic_ros2/test/unit/test_ParticipantRegistry.cpp [hunk 1]
```diff
@@ -49,29 +49,35 @@ SCENARIO("Test idempotency of shape type")
...
 
-
-bool operator==(const rmf_traffic::Profile p1, const rmf_traffic::Profile p2)
+namespace rmf_traffic {
+bool operator==(const Profile p1, const Profile p2)
 {
   return rmf_traffic_ros2::convert(p1) == rmf_traffic_ros2::convert(p2);
 }
-
+namespace schedule {
```
Comment (gbiggs) [inline rmf_traffic_ros2/test/unit/test_ParticipantRegistry.cpp:57]: Doesn't the linter want a new line after the namespace declaration and before its closing bracket?

Comment (arjo129) [reply rmf_traffic_ros2/test/unit/test_ParticipantRegistry.cpp:57]: Seems like the linter skips over namespaces. I'll go ahead and do that.

**Review by gbiggs — APPROVED** [review]

**Pull request merged**